### DISCUSSION TITLE
Want to find a way to set OPENID parameters in environment variables or .yml files

### DISCUSSION THREAD

**Discussion by guancyxx** [Budibase/budibase#9562] [Help]
I want to make budibase runing as an application on [arkid](https://github.com/longguikeji/arkid) (an open source IDAAS service we are developing) through the OIDC protocol. In this scenario, I need to configure openid-related content (such as client_id, etc.) through environment variables or configuration files, but I found that it seems It can only be set through the user interface. Is there an existing way for me to realize my idea or can you guide me how to modify the source code?

**Comment by melohagan** [comment]
Hey @guancyxx 

Currently this can't be set in environment variables for OIDC, but I can see there is an existing related issue open here: https://github.com/Budibase/budibase/issues/8466

In the meantime, you could look at how the Google OAuth config is done if you wanted to try and implement it yourself: https://github.com/Budibase/budibase/commit/096672098214e389d467a4aae6cfb2c8a56fe0d4

There's a 'passport' file for **oidc.js**: https://github.com/Budibase/budibase/tree/096672098214e389d467a4aae6cfb2c8a56fe0d4/packages/backend-core/src/middleware/passport
